### PR TITLE
Use a custom DOM element and template for output cells.

### PR DESCRIPTION
**Pull request by smashwilson**
[The future is now](http://www.html5rocks.com/en/tutorials/webcomponents/customelements/) :rocket: :stars:

- [x] Refactor browser process JS to an external script
- [x] Move kernel IO message handling to the frontend

### CODE REVIEW

**Comment by smashwilson** [comment]
Current status:

screen shot 2015-06-26 at 4 01 56 pm

**Comment by rgbkrk** [comment]
Whoa nifty.

**Comment by smashwilson** [comment]
@rgbkrk: Here's the [import document](http://www.html5rocks.com/en/tutorials/webcomponents/imports/) I'm telling you about literally right now.

**Comment by smashwilson** [comment]
This one's ready to go :tada:

**Comment by rgbkrk** [comment]
Something weird is going on with `master` here:

`Unknown system error, uv_pipe_open`

> it seems that the issue, at least for me, is induced by opening sockets with node.js net in the main app process

Did the refactor cause issues with this bit?

https://github.com/atom/atom/issues/6632
https://github.com/atom/electron/issues/1429

File: sidecar.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+var ipc = require('ipc');
+var jupyter = require('./lib/jupyter');
+
+var display = new jupyter.RichDisplay();
+
+function appendNode(elementName, html) {
+  var node = document.createElement(elementName);
+  node.setPayload(html);
+  container.appendChild(node);
```
Comment (rgbkrk) [inline sidecar.js]: You just blew my mind. I didn't realize you could access elements by ID directly like this until now.

**Pull request merged**